src/serializers/type_serializers/datetime_etc.rs

```diff
@@ -3,19 +3,27 @@ use std::borrow::Cow;
 use pyo3::prelude::*;
 use pyo3::types::{PyDate, PyDateTime, PyDict, PyTime};
 
-use crate::definitions::DefinitionsBuilder;
-use crate::input::{pydate_as_date, pydatetime_as_datetime, pytime_as_time};
-use crate::PydanticSerializationUnexpectedValue;
-
 use super::{
     infer_json_key, infer_serialize, infer_to_python, py_err_se_err, BuildSerializer, CombinedSerializer, Extra,
     SerMode, TypeSerializer,
 };
+use crate::definitions::DefinitionsBuilder;
+use crate::input::{pydate_as_date, pydatetime_as_datetime, pytime_as_time};
+use crate::serializers::config::{DatetimeMode, FromConfig};
+use crate::PydanticSerializationUnexpectedValue;
 
 pub(crate) fn datetime_to_string(py_dt: &Bound<'_, PyDateTime>) -> PyResult<String> {
     pydatetime_as_datetime(py_dt).map(|dt| dt.to_string())
 }
 
+pub(crate) fn datetime_to_seconds(py_dt: &Bound<'_, PyDateTime>) -> PyResult<i64> {
+    pydatetime_as_datetime(py_dt).map(|dt| dt.timestamp())
```

Comment:
Hm, seems to be an issue here :(

Looks like dt.timestamp() here doesn't take into account the number of microseconds in the datetime, so anything under a second resolution will get ignored.

Looks like its a speed date thing:
https://github.com/pydantic/speedate/blob/d15ac62b7e98d6e4b8a769b6f5ffab2ad1db5ca5/src/datetime.rs#L580-L582
https://github.com/pydantic/speedate/blob/main/src/time.rs#L388-L393

So i guess a few options:
1. We don't care, ignore (i dont think this is right)
2. We care and can calculate the timestamp ourselves here
3. We care and need to add something upstream to speeddate to allow for returning the timestamp as a f64 with microsecond precision

🤷🏻